### PR TITLE
Bug 1498954 - Broker in developer mode must support apb push

### PR DESCRIPTION
**Pull request by djzager**
# Testing for this PR

Just used the apb tool to push an image to the broker and looked at how the image was stored by looking at the logs of the broker.

## Before Change:

```
[2017-10-05T14:22:55.126Z] [DEBUG] handler::apbAddSpec
[2017-10-05T14:22:55.126Z] [DEBUG] Successfully decoded pushed spec:
[2017-10-05T14:22:55.126Z] [DEBUG] version: 1.0
name: hello-world-apb
description: deploys hello-world web application
bindable: False
async: optional
metadata:
  displayName: Hello World (APB)
  longDescription: A sample APB which deploys a containerized Hello World web application
  dependencies: ['docker.io/ansibleplaybookbundle/hello-world:latest']
  providerDisplayName: "Red Hat, Inc."
plans:
  - name: default
    description: A sample APB which deploys Hello World
    free: True
    metadata:
      displayName: Default
      longDescription: This plan deploys a Python web application displaying Hello World
      cost: $0.00
    parameters: []
 
[2017-10-05T14:22:55.127Z] [DEBUG] Unmarshalled into apb.Spec:
[2017-10-05T14:22:55.127Z] [DEBUG] {ID: Version:1.0 FQName:hello-world-apb Image: Tags:[] Bindable:false Description:deploys hello-world web application Metadata:map[displayName:Hello World (APB) longDescription:A sample APB which deploys a containerized Hello World web application dependencies:[docker.io/ansibleplaybookbundle/hello-world:latest] providerDisplayName:Red Hat, Inc.] Async:optional Plans:[{Name:default Description:A sample APB which deploys Hello World Metadata:map[displayName:Default longDescription:This plan deploys a Python web application displaying Hello World cost:$0.00] Free:true Bindable:false Parameters:[]}]}
```

## After Change:

```
[2017-10-05T15:23:16.817Z] [DEBUG] handler::apbAddSpec
[2017-10-05T15:23:16.817Z] [DEBUG] Successfully decoded pushed spec:
[2017-10-05T15:23:16.817Z] [DEBUG] version: 1.0
name: hello-world-apb
description: deploys hello-world web application
bindable: False
async: optional
metadata:
  displayName: Hello World (APB)
  longDescription: A sample APB which deploys a containerized Hello World web application
  dependencies: ['docker.io/ansibleplaybookbundle/hello-world:latest']
  providerDisplayName: "Red Hat, Inc."
plans:
  - name: default
    description: A sample APB which deploys Hello World
    free: True
    metadata:
      displayName: Default
      longDescription: This plan deploys a Python web application displaying Hello World
      cost: $0.00
    parameters: []
 
[2017-10-05T15:23:16.817Z] [DEBUG] Unmarshalled into apb.Spec:
[2017-10-05T15:23:16.818Z] [DEBUG] {ID: Version:1.0 FQName:hello-world-apb Image:hello-world-apb Tags:[] Bindable:false Description:deploys hello-world web application Metadata:map[displayName:Hello World (APB) longDescription:A sample APB which deploys a containerized Hello World web application dependencies:[docker.io/ansibleplaybookbundle/hello-world:latest] providerDisplayName:Red Hat, Inc.] Async:optional Plans:[{Name:default Description:A sample APB which deploys Hello World Metadata:map[displayName:Default longDescription:This plan deploys a Python web application displaying Hello World cost:$0.00] Free:true Bindable:false Parameters:[]}]}
```

### CODE REVIEW

**Comment by djzager** [comment]
@karmab If you were able to test this change for your purposes, I would appreciate it.

File: pkg/handler/handler.go [hunk 1]
```diff
@@ -564,6 +564,7 @@ func (h handler) apbAddSpec(w http.ResponseWriter, r *http.Request, params map[s
 		writeResponse(w, http.StatusBadRequest, broker.ErrorResponse{Description: "Invalid parameter yaml"})
 		return
 	}
+	spec.Image = spec.FQName
```
Comment (jmrodri) [inline pkg/handler/handler.go]: Feels weird to put this in the handler. I would put it in `AddSpec` of `broker.go`.

**Comment by karmab** [comment]
@djzager thanks for the change, do i have to build everything in order to test this one?

**Comment by karmab** [comment]
i tested the PR and didnt get the error, but i m getting
[INFO] error: unable to upgrade connection: container not found ("apb")
so the apb pod doesnt start and no provisioning occurs, neither for my apb or for any

full traceback at https://pastebin.com/Jw5fzUaK

File: pkg/handler/handler.go (hunk 1 above)
Comment (dymurray) [reply pkg/handler/handler.go]: +1 putting it in AddSpec makes more sense because that's what gets called when you push to the dev endpoint.

**Pull request merged**